app/src/main/java/nu/yona/app/api/manager/impl/NotificationManagerImpl.java

```diff
@@ -161,6 +162,87 @@ public void onError(Object errorMessage) {
         }
     }
 
+
+    public void getMessageWithUrl(final String urlForMessagesFetch, final boolean isUnreadStatus, final DataLoadListener listener, final boolean isProcessUpdate) {
+        try {
+            User user = YonaApplication.getEventChangeManager().getDataState().getUser();
+            if (user != null && user.getLinks() != null && user.getLinks().getYonaMessages() != null
+                    && !TextUtils.isEmpty(user.getLinks().getYonaMessages().getHref())) {
+
+                notificationNetwork.getNextSetOfMessagesFromURL(urlForMessagesFetch, YonaApplication.getEventChangeManager().getSharedPreference().getYonaPassword(), isUnreadStatus, new DataLoadListener() {
+                    @Override
+                    public void onDataLoad(Object result) {
+                        if (listener != null) {
+                            if (result instanceof YonaMessages) {
+                                YonaMessages resultYonaMessagesObj = (YonaMessages) result;
+                                if (resultYonaMessagesObj != null && resultYonaMessagesObj.getEmbedded() != null) {
```

Comment:
Can you realistically be called with a null result?

Reply:
Nope. I will update and remove the check and test it.